src/engine/routing_algorithms/alternative_path_mld.cpp

```diff
@@ -0,0 +1,786 @@
+#include "engine/routing_algorithms/alternative_path.hpp"
+#include "engine/routing_algorithms/routing_base_mld.hpp"
+
+#include "util/static_assert.hpp"
+
+#include <boost/assert.hpp>
+
+#include <algorithm>
+#include <iterator>
+#include <memory>
+#include <type_traits>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+#include <boost/function_output_iterator.hpp>
+
+// TODO: debug, remove
+#include <iostream>
+
+namespace osrm
+{
+namespace engine
+{
+namespace routing_algorithms
+{
+
+// Unqualified calls below are from the mld namespace.
+// This alternative implementation works only for mld.
+using namespace mld;
+
+using Heap = SearchEngineData<Algorithm>::QueryHeap;
+using Partition = partition::MultiLevelPartitionView;
+using Facade = datafacade::ContiguousInternalMemoryDataFacade<Algorithm>;
+
+// Implementation details
+namespace
+{
+
+// Alternative paths candidate via nodes are taken from overlapping search spaces.
+// Overlapping by a third guarantees us taking candidate nodes "from the middle".
+const constexpr auto kSearchSpaceOverlapFactor = 1.66;
+// Maximum number of alternative paths to return.
+const constexpr auto kMaxAlternatives = 3;
```

Comment:
Not sure if it should be a query-time parameter but I can definitely see the point in exposing some knobs here in the profiles.